js/researcher.js

```diff
@@ -0,0 +1,92 @@
+var Paper = require( "./values/Paper.js" );
+var merge = require( "lodash/object/merge" );
+var wordCount = require( "./stringProcessing/countWords.js" );
+var InvalidTypeError = require( "./errors/invalidType" );
+var MissingArgument = require( "./errors/missingArgument" );
+var isUndefined = require( "lodash/lang/isUndefined" );
+var isEmpty = require( "lodash/lang/isEmpty" );
+
+/**
+ * This contains all possible, default researches.
+ * @param {Paper} paper The Paper object that is needed within the researches.
+ * @constructor
+ * @throws {InvalidTypeError} Parameter needs to be an instance of the Paper object.
+ */
+var Researcher = function( paper ) {
+	this.updatePaper( paper );
+
+	this.defaultResearches = {
+		"wordCount": wordCount
+	};
+
+	this.customResearches = {};
+};
+
+/**
+ * Update the Paper associated with the Researcher.
+ * @param {Paper} paper The Paper to use within the Researcher
+ * @throws {InvalidTypeError} Parameter needs to be an instance of the Paper object.
+ */
+Researcher.prototype.updatePaper = function( paper ) {
```

Comment:
You're not 'validating' anything here. What happens is that the Paper object gets updated whenever text has been updated in the interface. Because the Researcher isn't continuously updated, this method was introduced.